### PR TITLE
fix: add tests, fix bug that was caught through tests

### PR DESCRIPTION
**Pull request by willKim19**

### CODE REVIEW

File: src/mapping.ts [hunk 1]
```diff
@@ -42,9 +47,10 @@ import { User, Pod } from "../generated/schema"
 // }
 
 export function handleTransferSingle(event: TransferSingle): void {
-  const to = event.params.to;
-  const from = event.params.from;
-  const id = event.params.id;
+  let to = event.params.to;
+  let from = event.params.from;
+  let id = event.params.id;
```
Comment (jtlacci) [inline src/mapping.ts:52]:
```js 
let to = event.params.to.toHex();
let from = event.params.from.toHex();
let id = event.params.id.toString();
```

File: src/mapping.ts [hunk 2]
```diff
@@ -54,21 +60,21 @@ export function handleTransferSingle(event: TransferSingle): void {
...
 
-  const fromUser = User.load(from.toHex());
+  let fromUser = User.load(from.toHex());
   // The fromUser should never be null, because the User is created
   // when they first receive a token.
   // This check is just assurance.
   if (fromUser !== null) {
     // Remove the pod from the user's "pods" array
-    fromUser.pods = fromUser.pods.splice(fromUser.pods.indexOf(id.toString()), 1);
+    fromUser.pods = removeItem(fromUser.pods, id.toString());
```
Comment (jtlacci) [inline src/mapping.ts:69]: i think we can replace this with a lodash function
```js
fromUser.pods = _.remove(fromUser.pods, elem => elem === id.toString());
```

File: src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ * @param value 
+ * @returns 
+ */
+export function removeItem(arr: Array<string> | null, value: string): Array<string> | null { 
+  let index = arr.indexOf(value);
+  if (index > -1) {
+      arr.splice(index, 1);
+  }
+  return arr;
+}
```
Comment (jtlacci) [inline src/utils.ts:13]: replace file w lodash

File: test/eventGenerators.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  let TransferSingleEvent = new TransferSingle();
+
+  TransferSingleEvent.parameters.push(AddressParam(operator));
+  TransferSingleEvent.parameters.push(AddressParam(from));
+  TransferSingleEvent.parameters.push(AddressParam(to));
+  TransferSingleEvent.parameters.push(I32Param(id));
+  TransferSingleEvent.parameters.push(I32Param(value));
+
+  return TransferSingleEvent;
+}
```
Comment (jtlacci) [inline test/eventGenerators.ts:40]: new line

File: test/eventGenerators.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+export function generateTransferSingle(operator: string, from: string, to: string, id: i32, value: i32): TransferSingle {
+  let TransferSingleEvent = new TransferSingle();
+
+  TransferSingleEvent.parameters.push(AddressParam(operator));
+  TransferSingleEvent.parameters.push(AddressParam(from));
+  TransferSingleEvent.parameters.push(AddressParam(to));
+  TransferSingleEvent.parameters.push(I32Param(id));
+  TransferSingleEvent.parameters.push(I32Param(value));
```
Comment (jtlacci) [inline test/eventGenerators.ts:37]:
```js
TransferSingleEvent.parameters.push(
  AddressParam(operator),
  AddressParam(from),
  AddressParam(to),
  I32Param(id),
  I32Param(value),
);
```

File: test/integration.spec.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import { clearStore, test, assert } from "matchstick-as/assembly/index";
+import { handleCreateSafe, handleTransferSingle } from "../src/mapping";
+import { generateCreateSafe, generateTransferSingle } from "./eventGenerators";
+import { log } from "matchstick-as/assembly/log";
+import { User } from "../generated/schema";
+
+const addressZero = '0x0000000000000000000000000000000000000000';
```
Comment (jtlacci) [inline test/integration.spec.ts:7]: `ethers.constants.AddressZero`

File: test/integration.spec.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      1,
+    );
+    
+    handleTransferSingle(transferSingle);
+    assert.fieldEquals('User', addressOne, 'pods', '[2, 3]');
+    assert.fieldEquals('User', addressTwo, 'pods', '[1]');
+    clearStore();
+    transferSingle.parameters.splice(0, 100);
+  }); 
+}
```
Comment (jtlacci) [inline test/integration.spec.ts:59]: newline

**Review by jtlacci — CHANGES_REQUESTED** [review]

File: src/mapping.ts (hunk 2 above)
Comment (willKim19) [reply src/mapping.ts:69]: This code snippet is going to be removed with my next PR anyway so lets just keep it for now

File: test/integration.spec.ts (hunk 6 above)
Comment (willKim19) [reply test/integration.spec.ts:7]: Can't import ethers as this is technically assemblyscript

File: test/eventGenerators.ts (hunk 5 above)
Comment (willKim19) [reply test/eventGenerators.ts:37]: Doesn't work in assemblyscript

**Pull request merged**